go/parser/parser.go

```diff
@@ -2027,8 +2027,12 @@ func parseTypeSpec(p *parser, doc *ast.CommentGroup, decl *ast.GenDecl, _ int) a
 	// at the identifier in the TypeSpec and ends at the end of the innermost
 	// containing block.
 	// (Global identifiers are resolved in a separate phase after parsing.)
-	spec := &ast.TypeSpec{doc, ident, nil, p.lineComment}
+	spec := &ast.TypeSpec{doc, ident, 0, nil, p.lineComment}
```

Comment:
this is not enough, should add some line to parse rightly

```go
+	spec := &ast.TypeSpec{doc, ident, token.NoPos, nil, p.lineComment}
 	p.declare(spec, p.topScope, ast.Typ, ident)
+	if p.tok == token.ASSIGN {
+		spec.Assign = p.pos
+		p.next()
+	}
```

Reply:
I must be missing something, apart from the `token.NoPos` change (which I've just made), your proposal looks identical to what is already there to me.

Reply:
my fault, confused with the pull-request page ;)